deploy/template.go

```diff
@@ -247,14 +259,41 @@ func (c Controller) create(client Client, template *Template, nameSpace string,
 	return complete, nil
 }
 
-func (c Controller) update(client Client, dc *dc.DeploymentConfig, template *Template, nameSpace string, deploy *Payload) (*Complete, error) {
+// update the existing deployconfig and buildconfig with the new deployment payload data
+func (c Controller) update(client Client, d *dc.DeploymentConfig, b *bc.BuildConfig, template *Template, nameSpace string, deploy *Payload) (*Complete, error) {
 	var (
 		complete = &Complete{}
 	)
-	// for _, c := range dc.Spec.Template.Spec.Containers{
-	// 	c.Env
-	// }
-	//update git details
-	// update env vars
-	return complete, errors.New("redeploy not implemented")
+
+	// 03/03/2017 pbrookes: Iterate over the objects in the template and update the ones we care about
+	for _, ob := range template.Objects {
```

Comment:
Interesting. Could you help me understand why we use the template again instead of just pulling back and updating the specific objects directly from OSCP?